src/components/FlawHistory.vue

```diff
@@ -1,17 +1,36 @@
 <script setup lang="ts">
-import { ref } from 'vue';
+import { computed, ref } from 'vue';
+
+import { usePagination } from '@/composables/usePagination';
 
 import { capitalize, formatDateWithTimezone } from '@/utils/helpers';
 import { flawFieldNamesMapping } from '@/constants/flawFields';
-import type { ZodFlawType } from '@/types/zodFlaw';
+import type { ZodFlawHistoryItemType } from '@/types/zodFlaw';
 
 import LabelCollapsible from './widgets/LabelCollapsible.vue';
 
-defineProps<{
-  flaw: ZodFlawType;
+const props = defineProps<{
+  history: null | undefined | ZodFlawHistoryItemType[];
 }>();
 
-const historyExpanded = ref(false);
+const historyExpanded = ref(true);
+
+const itemsPerPage = 20;
+const totalPages = computed(() =>
+  Math.ceil((props.history?.length || 0) / itemsPerPage),
+);
+
+const {
+  changePage,
+  currentPage,
+  pages,
+} = usePagination(totalPages, itemsPerPage);
+
+const paginatedHistoryItems = computed(() => {
+  const start = (currentPage.value - 1) * itemsPerPage;
+  const end = start + itemsPerPage;
+  return props.history?.slice(start, end);
+});
```

Comment:
We need to do this exact calculation each time we use the `usePagination` composable, maybe is worth modifying the composable so you pass in the array of elements and it calculates the `totalPages` and also the `paginatedItems` directly.

https://github.com/RedHatProductSecurity/osim/blob/48740f7834538a7ce11e44762d4966b5a878300e/src/composables/usePaginationWithSettings.ts#L48-L52

Reply:
As discussed on meeting this will be handled in an independent task/PR (OSIDB-36679)